test/ts/unit/simple_interest_terms_contract.ts

```diff
@@ -356,6 +370,15 @@ contract("SimpleInterestTermsContract (Unit Tests)", async (ACCOUNTS) => {
                 );
             });
 
+            it("unpacks valid params", async () => {
```

Comment:
I think it's good you've added this test, but it should probably be under a different scope --> right now it's within the `getExpectedRepaymentValue` scope, whereas it should probably be under an individual `#unpackParametersFromBytes` scope.

Reply:
Agreed